### PR TITLE
Fixes on audit comments.

### PR DESCRIPTION
**Pull request by drortirosh**
following Zepelin's audit added some fixes.

fixed pragma solidity
removed abstract methods from RelayRecipient (as they duplicate IRelayRecipient's)
added documentation of main methods.
cleaned up comments

removed initRelayHub() - to signify setRelayHub should be used (and support updated relay)

### CODE REVIEW

File: contracts/RelayHub.sol [hunk 1]
```diff
@@ -143,7 +144,7 @@ contract RelayHub is IRelayHub {
         require(relays[relay].owner == msg.sender, "not owner");
         relays[relay].unstakeTime = relays[relay].unstakeDelay + now;
         // Start the unstake counter
-        if (relays[relay].state != State.PENALIZED) {
+        if (relays[relay].state != State.PENALIZED && relays[relay].state != State.REMOVED) {
```
Comment (shahafn) [inline contracts/RelayHub.sol]: Not sure what the `relays[relay].state != State.REMOVED` check achieves.

Comment (drortirosh) [reply contracts/RelayHub.sol]: Without it, calling remove again will re-trigger the unstake timeout...

Comment (shahafn) [reply contracts/RelayHub.sol]: unstake timeout is set right before this `if`, this branch merely  sets `relays[relay].state = State.REMOVED`.

Comment (shahafn) [inline contracts/RelayHub.sol]: make this a require in the beginning of function

File: contracts/RelayHub.sol [hunk 2]
```diff
@@ -328,7 +328,12 @@ contract RelayHub is IRelayHub {
         require(relays[addr1].stake > 0, "Unstaked relay");
         // Checking that the relay wasn't penalized yet
         require(relays[addr1].state != State.PENALIZED, "Relay already penalized");
-        // compensating the sender with the stake of the relay
+
+        // compensating the sender with HALF the stake of the relay (the other half is burned)
+        uint to_burn = relays[addr1].stake/2;
```
Comment (shahafn) [inline contracts/RelayHub.sol]: 1. Please keep camelCase
2. Use safemath

**Pull request merged**